packages/utils/src/revert_error.ts

```diff
@@ -408,10 +410,34 @@ function declarationToAbi(declaration: string): RevertErrorAbi {
 function checkArgEquality(type: string, lhs: ArgTypes, rhs: ArgTypes): boolean {
     if (type === 'address') {
         return normalizeAddress(lhs as string) === normalizeAddress(rhs as string);
-    } else if (type.startsWith('bytes')) {
+    } else if (type === 'bytes' || /^bytes(\d+)$/.test(type)) {
         return normalizeBytes(lhs as string) === normalizeBytes(rhs as string);
     } else if (type === 'string') {
         return lhs === rhs;
+    } else if (/\[\d*\]$/.test(type)) {
```

Comment:
Very clean! Might be good to add a comment on the larger regexs, but they aren't too bad yet.